staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -36,9 +36,27 @@ type Checker interface {
 
 // NewChecker creates a new policy checker.
 func NewChecker(policy *audit.Policy) Checker {
+	for i, rule := range policy.Rules {
+		policy.Rules[i].OmitStages = unionStages(policy.OmitStages, rule.OmitStages)
+	}
 	return &policyChecker{*policy}
 }
 
+func unionStages(s1, s2 []audit.Stage) []audit.Stage {
```

Comment:
Why not

```go
func unionStages(stageLists ...[]audit.Stage) []audit.Stage {
  m := make(map[audit.Stage]bool)
  for _, sl := range stageLists {
    for _, s := range sl {
      m[s] = true
    }
  }
  result := make([]audit.Stage, 0, len(m))
  for key := range m {
      result = append(result, key)
  }
  return result
}
```

- Why do you need exactly two parameters in this case?
- map[T]bool is more frequently used approach than map[T]struct{}

Reply:
Done.